lib/Parse/ParseDecl.cpp

```diff
@@ -5943,6 +5943,17 @@ void Parser::ParseParameterDeclarationClause(
       // Inform the actions module about the parameter declarator, so it gets
       // added to the current scope.
       Decl *Param = Actions.ActOnParamDeclarator(getCurScope(), ParmDeclarator);
+
+      // Parse the bounds expression, if any.
+      if (getLangOpts().CheckedC && Tok.is(tok::colon)) {
+        ConsumeToken();
+        ExprResult Bounds = ParseBoundsExpression();
+        if (Bounds.isInvalid()) {
+          SkipUntil(tok::comma, tok::r_paren, StopAtSemi | StopBeforeMatch);
+          Actions.ActOnInvalidBoundsExpr(Param);
+        } else Actions.ActOnBoundsExpr(Param, cast<BoundsExpr>(Bounds.get()));
```

Comment:
Agreed.  I looked at the code and else statements have on the same line either nothing, {, or an if-control statement:
```
else
   ...
```
or
```
else { 
  ...
 }
```
or 
```
else if (cond) 
```